data_structures/stacks/stack.py

```diff
@@ -29,21 +26,24 @@ def push(self, data):
 
     def pop(self):
         """ Pop an element off of the top of the stack."""
-        if self.stack:
-            return self.stack.pop()
-        else:
+        if self.is_empty():
             raise IndexError("pop from an empty stack")
+        return self.stack.pop()
 
     def peek(self):
         """ Peek at the top-most element of the stack."""
-        if self.stack:
-            return self.stack[-1]
+        if self.is_empty():
+            raise IndexError("peek from an empty stack")
```

Comment:
Does it do a similar thing if you remove lines 35 and 36?

Reply:
Line3 35 and 36 execute when stack is empty.  `assert()` statements will still be successful if i remove them .